Source/radiation/MGFLD.cpp

```diff
@@ -1001,16 +1002,1244 @@ void Radiation::compute_limiter(int level, const BoxArray& grids,
+                }
+                else {
+                    r3 = (Er(i,j,k+1,g) - Er(i,j,k-1,g)) / (2.e0_rt*dx[2]);
+                }
+#endif
+
+                Real r = std::sqrt(r1 * r1 + r2 * r2 + r3 * r3);
+                r = r / (kap(i,j,k,g) * std::max(Er(i,j,k,g), 1.e-50_rt));
+
+                lam(i,j,k,g) = FLDlambda(r);
+            });
+
+            // filter
+
+            int lam_ilo = lambx.loVect3d()[0];
+            int lam_ihi = lambx.hiVect3d()[0];
+            int lam_jlo = lambx.loVect3d()[1];
+            int lam_jhi = lambx.hiVect3d()[1];
+            int lam_klo = lambx.loVect3d()[2];
+            int lam_khi = lambx.hiVect3d()[2];
+
+            int reg_ilo = bx.loVect3d()[0];
+            int reg_ihi = bx.hiVect3d()[0];
+            int reg_jlo = bx.loVect3d()[1];
+            int reg_jhi = bx.hiVect3d()[1];
+            int reg_klo = bx.loVect3d()[2];
+            int reg_khi = bx.hiVect3d()[2];
+
+            if (filter_lambda_T == 1) {
+
+                for (int k = lam_klo; k <= lam_khi; ++k) {
+                    for (int j = lam_jlo; j <= lam_jhi; ++j) {
+#if AMREX_SPACEDIM >= 2
+                        if (Er(reg_ilo,j,k,g) == -1.e0_rt) {
+                            for (int i = reg_ilo; i <= reg_ihi; ++i) {
+                                lamfil(i,j,k) = -1.e-50_rt;
+                            }
+                            continue;
+                        }
+#endif
+
+                        for (int i = reg_ilo; i <= reg_ihi; ++i) {
+                            lamfil(i,j,k) = ff1(0) * lam(i,j,k,g) + ff1(1) * (lam(i-1,j,k,g) + lam(i+1,j,k,g));
+                            lamfil(i,j,k) = std::min(1.e0_rt/3.e0_rt, std::max(1.e-25_rt, lamfil(i,j,k)));
+                        }
+
+                        if (Er(reg_ilo-1,j,k,g) == -1.e0_rt) {
+                            int i = reg_ilo;
+                            lamfil(i,j,k) = ff1b(0) * lam(i,j,k,g) + ff1b(1) * lam(i+1,j,k,g);
+                            lamfil(i,j,k) = std::min(1.e0_rt/3.e0_rt, std::max(1.e-25_rt, lamfil(i,j,k)));
+                        }
+
+                        if (Er(reg_ihi+1,j,k,g) == -1.e0_rt) {
+                            int i = reg_ihi;
+                            lamfil(i,j,k) = ff1b(1) * lam(i-1,j,k,g) + ff1b(0) * lam(i,j,k,g);
+                            lamfil(i,j,k) = std::min(1.e0_rt/3.e0_rt, std::max(1.e-25_rt, lamfil(i,j,k)));
+                        }
+                    }
+                }
+
+#if AMREX_SPACEDIM >= 2
+                for (int k = lam_klo; k <= lam_khi; ++k) {
+                    if (Er(reg_ilo,reg_jlo,k,g) == -1.e0_rt) {
+                        continue;
+                    }
+
+                    for (int j = reg_jlo; j <= reg_jhi; ++j) {
+                        for (int i = reg_ilo; i <= reg_ihi; ++j) {
+                            lam(i,j,k,g) = ff1(0) * lamfil(i,j,k) + ff1(1) * (lamfil(i,j-1,k) + lamfil(i,j+1,k));
+                        }
```

Comment:
we seem to be missing the lines like:
```
lam(i,j,g) = min(1.e0_rt/3.e0_rt, max(1.e-25_rt, lam(i,j,g))) 
````
that appear in `CastroRad_2d.f90` in this section

Reply:
also, shouldn't we be fulling `lamfil` here, since we do a copy down below from lamfil to lam?

Reply:
Yes I agree that something was off here but this was another case where there was an inconsistency between 1D and 2D and 3D, so I used the 3D code as the backbone and picked an arbitrary convention where we applied that limiter to lamfil but not to lam (I know, there's no real justification for this). I am game for fixing this but since apparently no one has been using this for 2D or 3D anyway, I'm not sure what actually is. correct.